src/pip/_internal/commands/install.py

```diff
@@ -594,6 +594,61 @@ def get_lib_location_guesses(*args, **kwargs):
     return [scheme['purelib'], scheme['platlib']]
 
 
+def site_packages_writable(**kwargs):
+    return all(
+        test_writable_dir(d) for d in get_lib_location_guesses(**kwargs)
```

Comment:
I would wrap the result of `get_lib_location_guesses` in `set()` since in a lot of cases (venv, virtualenv, altinstall'd Unix Pythons) it will have 2 of the same path.

Reply:
Good point, thanks.